File: 3018-make-string-a-subsequence-using-cyclic-increments/3018-make-string-a-subsequence-using-cyclic-increments.cpp
class Solution {
public:
    bool canMakeSubsequence(string a, string b) {
        int m=a.size();
        int n=b.size();
        int i=0,j=0;
        int ans=0;
        while(i<m and j<n){
            if(a[i]==b[j]||(b[j]-a[i]==1||(a[i]=='z' and b[j]=='a'))){
                i++,j++;
                ans++;
            }
            else{
                i++;
            }
            // cout<<i<<" "<<j<<" "<<ans<<endl;
        }
        return ans==n;
    }
};

File: 3018-make-string-a-subsequence-using-cyclic-increments/3018-make-string-a-subsequence-using-cyclic-increments.java
class Solution {
    public boolean canMakeSubsequence(String str1, String str2) {
        if(str2.length()>str1.length()) return false;
        int p2 = 0;int p1 = 0;
        while(p2<str2.length()  && p1<str1.length()){
            int ascii = str1.charAt(p1);
            int ascii2 = str2.charAt(p2);
            if(ascii==ascii2 || ascii+1==ascii2  || (ascii==122 && ascii2==97)){
                p2++;
            }
            p1++;
        }
        if(p2>=str2.length()) return true;
        else return false;
    }
}